### PR TITLE
Signup: Create a new flow without the free plan, and with a skip button

### LINKED ISSUE #3274: AB Test: Offer plans only in signup
We want to try an AB test in the signup flow where we only offer paid plans to the user, and have them skip the selection if they want a free plan.

### PR DESCRIPTION
**Pull request by scruffian**
This pull request fixes #3274 by creating a signup flow at /start/upgrade which doesn't contain the free plan:

screen shot 2016-02-16 at 12 48 06
 
Pinging @breezyskies and @mikeshelton1503 for a design review...

#### Testing instructions

1. Run `git checkout fix/3274-paid-plans-only` and start your server
2. Open the [`Upgrade signup flow` page](http://calypso.localhost:3000/start/upgrade)
3. Check that you can sign up for the free plan by using the "skip" button
3. Check that you can sign up for a paid plan by selecting a plan, and being sent to /checkout
3. Check that you can purchase a domain in signup

#### Additional notes
This also contains a change to the flow at /start/plan, adding the domain step, now that free trials are disabled there.

#### Reviews

- [x] Code
- [x] Product
- [x] Design

### CODE REVIEW

File: client/signup/steps/paid-plans-with-skip/index.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PaidPlansOnlyStepComponent from 'signup/steps/paid-plans-only';
```
Comment (stephanethomas) [inline client/signup/steps/paid-plans-with-skip/index.jsx]: Do we really have to suffix component names with `Component`?

File: client/signup/steps/paid-plans-with-skip/index.jsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+module.exports = React.createClass( {
```
Comment (stephanethomas) [inline client/signup/steps/paid-plans-with-skip/index.jsx]: We could use `export default` here.

**Comment by fabianapsimoes** [comment]
Product :+1:

File: client/signup/steps/paid-plans-with-skip/style.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.paid-plans-with-free-trials .plan-list {
```
Comment (stephanethomas) [inline client/signup/steps/paid-plans-with-skip/style.scss]: I'm a bit surprised that this selector starts with `.paid-plans-with-free-trials` and not `.paid-plans-with-skip` (which matches the name of the component). What's the reason?

File: client/signup/steps/plans/index.jsx [hunk 4]
```diff
@@ -162,6 +163,14 @@ module.exports = React.createClass( {
...
 				: this.plansSelection()
 			}
+			{
+				this.props.showSkipStepButton
+					? <SkipStepButton
+						goToNextStep={ this.onSelectPlan }
+						stepName={ this.props.stepName }
+						defaultDependencies={ this.props.defaultDependencies } />
+					: null
+			}
```
Comment (stephanethomas) [inline client/signup/steps/plans/index.jsx]: You could opt for a shorter approach:

```javascript
{
  this.props.showSkipStepButton &&
    <SkipStepButton
      goToNextStep={ this.onSelectPlan }
      stepName={ this.props.stepName }
      defaultDependencies={ this.props.defaultDependencies } />
}
```

**Comment by stephanethomas** [comment]
Could we take advantage of this pull request to fix the wrong [display name](https://github.com/Automattic/wp-calypso/blob/08d21c703334810e43bedf128f69842313322fe6/client/signup/steps/paid-plans-only/index.jsx#L12) of the `<PaidPlansOnlyStep />` component?

File: client/signup/steps/plans/index.jsx [hunk 5]
```diff
@@ -16,6 +16,7 @@ var productsList = require( 'lib/products-list' )(),
 	PlanList = require( 'components/plans/plan-list' ),
 	PlansCompare = require( 'components/plans/plans-compare' ),
 	SignupActions = require( 'lib/signup/actions' ),
+	SkipStepButton = require( 'signup/skip-step-button' ),
```
Comment (stephanethomas) [inline client/signup/steps/plans/index.jsx]: Could we use the skip feature that is available in the [`<StepWrapper />` component](https://github.com/Automattic/wp-calypso/blob/08d21c703334810e43bedf128f69842313322fe6/client/signup/step-wrapper/index.jsx)?

**Comment by mikeshelton1503** [comment]
Everything looks good except for the alignment of the back and skip buttons.

Lets inline and vertically center them:

image

I'll note that on the compare plans page we still show the skip button but no back button. I think this is good though since there is a back button at the top and the back button would mean something different if at the bottom.

File: client/signup/steps/plans/index.jsx (hunk 5 above)
Comment (scruffian) [inline client/signup/steps/plans/index.jsx]: Yes, that's what I did in aac8d71, good spot :)

File: client/signup/steps/paid-plans-with-skip/style.scss (hunk 3 above)
Comment (scruffian) [inline client/signup/steps/paid-plans-with-skip/style.scss]: We don't need this file at all.

**Comment by scruffian** [comment]
@stephanethomas thanks for the review. I (believe) I have address all of your points.

**Comment by danhauk** [comment]
I may have missed some discussion, but it looks a little odd as a button IMO. Is the intention there to call more attention to it? 

What if we matched the style of the "Back" link? It's not as prominent, but making it a longer string "Skip for now" or "Don't upgrade yet" will differentiate it from the back link.

image

**Comment by scruffian** [comment]
@danhauk would you make this suggestion for the other skip buttons in the flow too? (For example on the theme step?

**Comment by danhauk** [comment]
@scruffian The other ones (just one? I didn't see any others) don't look out of place, but I think keeping them all uniform is best. So if we go with a text link, change the others too.

**Comment by breezyskies** [comment]
+1 for the text link @danhauk suggested, including making these uniform throughout the flow.

File: client/signup/step-wrapper/style.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
+.step-wrapper {
+	.step-wrapper_buttons {
```
Comment (drewblaisdell) [inline client/signup/step-wrapper/style.scss]: Super minor, but:

- I don't think you need this extra class. Selecting with `.step-wrapper .skip-step` should be sufficient.
- If you do add it, it should contain two underscores, like `.step-wrapper__buttons`.

**Comment by drewblaisdell** [comment]
I added two commits with some minor updates, as well as a commit that changes the skip button to a text link by converting `PreviousStepButton` into `NavigationLink`, which now handles the skip case as well.

**Comment by scruffian** [comment]
Thanks for the commits @drewblaisdell. I wonder if, rather than having one component that handles both back and forward buttons, would the code be cleaner if we kept it as two components composed from a third component? (Not worth it for this PR, but maybe a future refactoring?)

**Comment by danhauk** [comment]
:+1: design

**Comment by fabianapsimoes** [comment]
Product :+1:

**Comment by drewblaisdell** [comment]
> I wonder if, rather than having one component that handles both back and forward buttons, would the code be cleaner if we kept it as two components composed from a third component?

@scruffian It might be also clearer if we went in the opposite direction, removing the component entirely and adding it to `StepWrapper` - it isn't used anywhere else, and `StepWrapper` is already a short function. This is a case where I think creating additional components makes the code slightly more difficult to follow than simply adding another anchor tag and event handler to a component.

**Comment by mikeshelton1503** [comment]
>I may have missed some discussion, but it looks a little odd as a button IMO. Is the intention there to call more attention to it?

@danhauk I was just following the Skip button that was already used in sign-up (on Themes step), but agree with your suggestion here to make them text links.

**Pull request merged**